dts: st: stm32h7rs: Add fdcan1 and fdcan2 configuration

TD-JBL opened the pull request:
This PR adds FDCAN support for STM32H7RS series microcontrollers.
As the H7RSx doesn't have the clock calibration unit (CCU) inside the FDCAN peripheral and a different message RAM size compared to H7x, I took the H5x configuration as template for this change. According to the reference manuals, the FDCAN peripheral is the same (except for register addresses, clock source and interrupt numbers). 

## Changes:
SoC support: Add FDCAN1 and FDCAN2 controller configurations for STM32H7RS series
- Includes register addresses, clock sources and interrupt definitions
- Board Support: Add FDCAN1 to NUCLEO_H7S3L8 board. PD0 (RX) and PD1 (TX) are used for interfacing an external CAN transceiver according to the breakout connector description in the board's schematic.

## Clock Configuration:
I added the second (optional) clock to the configuration (called domain clock in the driver), even if the HSE is already the default clock source. But programmers can then see more quickly how they can adjust the clock source.

## Testing Status:
- Tested on NUCLEO-H7S3L8 for FDCAN1 with external Waveshare SN65HVD230 CAN transceiver with samples/drivers/can/babbling.
- Tested on NUCLEO-H7S3L8 for FDCAN2 with external Waveshare SN65HVD230 CAN transceiver with custom board definition and app.

TD-JBL committed:
dts: arm: st: h7rs: Add fdcan1 and fdcan2 configuration

Provide the soc configuration for fdcan1 and fdcan2 controllers.
This includes registers address, clocks and interrupt lines
details.

Signed-off-by: Thomas Decker <decker@jb-lighting.de>

etienne-lms reviewed (commented):
For consistency in commits header lines in the Git revisions history, could your change it:
```patch
-dts: st: stm32h7rs: Add fdcan1 and fdcan2 configuration
+dts: arm: st: h7rs: Add fdcan1 and fdcan2 configuration 
```

erwango commented on dts/arm/st/h7rs/stm32h7rs.dtsi:
```diff
@@ -497,6 +497,32 @@
 		};
 
+		fdcan1: can@4000a000 {
+			compatible = "st,stm32-fdcan";
```
There is also a `st,stm32h7-fdcan` compatible. Are you sure you picked-up the right one ?

erwango commented on dts/arm/st/h7rs/stm32h7rs.dtsi:
```diff
@@ -497,6 +497,32 @@
+			reg-names = "m_can", "message_ram";
+			/* common clock FDCAN 1 & 2 */
+			clocks = <&rcc STM32_CLOCK(APB1_2, 8U)>,
+				<&rcc STM32_SRC_HSE FDCAN_SEL(0)>;
```
It should be noted that on H7RS, HSE is the default clock and providing the kernel clock is required to be able to compute the CAN block rate. So I agree providing HSE might be wrong, but not providing it would lead to other errors.
One idea could be to not provide the kernel clock and get the driver issue a compilation error if not provided (unfortunately, we can't generate an error at dts level if only the kernel clock is missing)

TD-JBL replied:
I added the fdcan1-node to the nucleo_h7s3l8 board similar to the configuration of the nucleo_h723zg. There the clock parameter with the second clock setting is in the board's dts and overwrites the default with only one clock. Should I remove the second clock from the dts?

erwango replied:
Thinking about it again:
If HSE is not enabled, you'll actually et an error in the can driver when trying to get rate.
So it may not be that bad to put it that way.
In the other alternative (not adding the second clock) you'll be able to get the rate but you'll end up with wrong results which is harder to debug.

If I'm right, keeping the second clock would be better for the user.

erwango reviewed (changes requested):
Please enable the node at board level so we have a way to test it in CI

TD-JBL commented:
> Please enable the node at board level so we have a way to test it in CI

The NUCLEO-H7S3L8 does not have a CAN transceiver onboard. Is it useful to enable CAN on some random pins on the breakout headers for this board? The bigger STM32H7S78-DK also does not have a CAN receiver. I'm not aware of other H7RSx boards.

TD-JBL committed:
boards: arm: st: nucleo_h7s3l8: Add fdcan1

Enable support for FDCAN1 on nucleo_h7s3l8 board. PD0 (RX) and PD1 (TX)
are used for interfacing an external CAN transceiver according to the
breakout connector description in the board's schematic.

Signed-off-by: Thomas Decker <decker@jb-lighting.de>

etienne-lms commented on boards/st/nucleo_h7s3l8/nucleo_h7s3l8.dts:
```diff
@@ -166,6 +167,14 @@
+&fdcan1 {
+	pinctrl-0 = <&fdcan1_rx_pd0 &fdcan1_tx_pd1>;
+	pinctrl-names = "default";
+	clocks = <&rcc STM32_CLOCK(APB1_2, 8U)>,
```
```suggestion
	clocks = <&rcc STM32_CLOCK(APB1_2, 8)>,

etienne-lms reviewed (commented):
For commit "boards: arm: st: nucleo_h7s3l8: Add fdcan1":
Could you break lines (74char/line) in the commit message body?

etienne-lms commented:
In case it helps, you can file the Zephyr guidelines on commit message formatting at:
https://docs.zephyrproject.org/latest/contribute/guidelines.html#commit-message-guidelines

erwango commented on boards/st/nucleo_h7s3l8/nucleo_h7s3l8.dts:
```diff
@@ -166,6 +167,14 @@
+&fdcan1 {
+	pinctrl-0 = <&fdcan1_rx_pd0 &fdcan1_tx_pd1>;
+	pinctrl-names = "default";
+	clocks = <&rcc STM32_CLOCK(APB1_2, 8)>,
+		<&rcc STM32_SRC_HSE FDCAN_SEL(0)>;
```
Non blocking: Adding `clocks` here is optional, but it could be clearer to keep it indeed.
